dev/lint/run_iwyu.sh

```diff
@@ -22,15 +22,34 @@ mkdir -p /build/lint
 pushd /build/lint
 
 cmake -GNinja \
+      -DARROW_FLIGHT=ON \
+      -DARROW_GANDIVA=ON \
       -DARROW_PARQUET=ON \
       -DARROW_PYTHON=ON \
       -DCMAKE_CXX_FLAGS='-D_GLIBCXX_USE_CXX11_ABI=0' \
       -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
       /arrow/cpp
-# Make so that vendored bits are built
-ninja arrow_shared
 
 popd
 
+# Build IWYU for current Clang
```

Comment:
This should go into the container instead of building it on each `docker-compose run iwyu`.

Please factor out into a scripts, e.g. `ci/docker_install_iwyu.sh` and build `iwyu` in its [Dockerfile](https://github.com/apache/arrow/blob/master/dev/lint/Dockerfile) similarly like We install [conda](https://github.com/apache/arrow/blob/master/cpp/Dockerfile#L36-L42).

Reply:
I'll finish this

Reply:
See https://github.com/wesm/arrow/pull/8